### ISSUE TITLE
Question about Multipath

### ISSUE THREAD

**Issue by sidmadala** [Schniz/fnm#353]
Thanks for making a fast node version manager (nvm was bricking my startup time). I just had a question about the point of the multipath option. Why doesn't fnm just point to the `$FNM_DIR/node-versions/<version>/installation/bin` and context switch instead of making temporary directories?

**Comment by sidmadala** [comment]
At this point I think this is a bug because I have quite a few multishell symlinks appearing and it's a lot of clutter for using one Node version. I've pasted a photo of how many there are and this is only half of the shells created. Is there a way to just keep one folder? I also find it really unintuitive as to why we are pointing to a path name that isn't super explicit about what version of Node we are using.

<img width="1306" alt="Screen Shot 2021-01-03 at 10 50 27 AM" src="https://user-images.githubusercontent.com/22417141/103482894-83293c80-4db1-11eb-92e8-e48d4c384c2d.png">

**Comment by Schniz** [comment]
Hey @sidmadala and thanks for the nice issue. I understand that all these `fnm_multishell_*` symlinks are daunting, and I'll try to explain how the multi-shell thingy works. Maybe it worth writing up in a blog post, but this is for the future.

Unlike its famous alternatives, `fnm` is a binary — compared to `nvm` and `n` which are Bash scripts _(not binaries that run Bash!)_. It's a blessing and a curse: `fnm` works on every shell, including PowerShell and CMD on Windows. But it also adds a layer between your shell and fnm, so the only way fnm can communicate with your shell and switch Node versions is by changing something in a shared context between the binary and the shell — something like the file system.

Before Multi-shell was merged, what we had was `$FNM_DIR/current` which was a _symlink_ to the current Node version. All we had to do was to change `$FNM_DIR/current` to the Node version of your choice, and it was easy and fast 😄 : 

```
~/.fnm/current -> ~/.fnm/node-versions/VERSION/installation
```

When we wanted to have Multi-shell, which is to allow to choose a Node version for each shell — so Node version won't be global, se couldn't have one global place to store the symlink, so we had to introduce a new place to store the current symlink. A temp directory is a good place: it cleans up when the operating system wants to do that (or at boot time) and symlinks are not expensive, so I don't think we should care they're lying there in the temp directory:

```
/tmp/fnm_multishell_RANDOM1_TIMESTAMP1 -> ~/.fnm/node-versions/VERSION/installation
/tmp/fnm_multishell_RANDOM2_TIMESTAMP2 -> ~/.fnm/node-versions/VERSION/installation
...
```

To sum up, using the version name in the symlink won't be helpful — the shell won't know how to switch to it. The symlink name is practically `fnm_multishell_SHELL_ID` which is linked to a version.

I'd be happy to know if you have a better idea of how to make a "context switch". fnm's magic is all about these symlinks right now 😄 

Let me know if it makes any sense

**Comment by sidmadala** [comment]
Thanks for the response, it's a lot clearer now as to why we are symlinking. In reality, I have no idea how to go about context switching since I am new to Rust. Maybe we can have a `fnm clean` function or something similar to prune multishell symlinks older than a certain number of days. As you mention, however, the OS already cleans the tmp folder. I don't really mind the symlinks since it appears that package install locations are not affected.

**Comment by struggzard** [comment]
What If I don't need multi shell feature? Can I switch to work in good old easy and fast `$FNM_DIR/current` mode? In most time I only care about single node version for a work day and don't really juggling with multiple node versions in parallel. Did I chose wrong tool and I should just switch to nvm instead?

**Comment by Schniz** [comment]
@struggzard PRs welcome 🤗
